core/scripts/bareos-config-lib.sh.in

```diff
@@ -758,82 +720,119 @@ set_config_param_in_file()
   printed == 1 {
     printed = 0
   }
-    ' "$FILE" > "$TEMP"
+    ' "${FILE}" > "${TEMP}"
     if [ $? -ne 0 ]; then
         warn "failed to set parameter ${SECTION} ${NAME} ${PARAM} = ${VALUE}: replace failed"
-        rm "$TEMP"
+        rm -f "${TEMP}"
         return 1
     fi
 
-    chown --reference="${FILE}" "$TEMP"
-    chmod --reference="${FILE}" "$TEMP"
+    chown --reference="${FILE}" "${TEMP}"
+    chmod --reference="${FILE}" "${TEMP}"
 
-    mv "$TEMP" "$FILE"
-    if [ $? != 0 ]; then
-        warn "failed to set parameter ${SECTION} ${NAME} ${PARAM} = ${VALUE}: replacing $FILE failed"
-        rm -f "$TEMP"
+    if ! mv "${TEMP}" "${FILE}"
+    then
+        warn "failed to set parameter ${SECTION} ${NAME} ${PARAM} = ${VALUE}: replacing ${FILE} failed"
+        rm -f "${TEMP}"
         return 1
     fi
 }
 
-[ ${os_type} != Linux ] && \
-set_config_param()
-{
-   echo "set_config_param() is not supported on this platform"
-   exit 1
-}
-
-
 get_database_param()
 {
     PARAM="${1-}"
     DEFAULT="${2-}"
+    VALUE=""
 
-    # use || to prevent errors
-    rc=0
-    temp_log="/tmp/bareos-config.$$.log"
-    DBCHECK_OUTPUT=`$DBCHECK 2>> $temp_log` || rc=$?
-    if [ $rc != 0 ]; then
-        echo "${DBCHECK_OUTPUT}" >> $temp_log
-        echo "Error: executing $DBCHECK" >> $temp_log
-        echo "" >> $temp_log
-
+    # Check if PARAM is valid, export is important to be catched by awk
+    case "${PARAM}" in
+        "catalog")
+            export NEEDLE="Name"
+            ;;
+        "dbdriver")
+            echo "postgresql"
+            return 0
+            ;;
+        "dbaddress")
+            export NEEDLE="DbAddress"
+            ;;
+        "dbsocket")
+            export NEEDLE="DbSocket"
+            ;;
+        "dbport")
+            export NEEDLE="DbPort"
+            ;;
+        "dbname")
+            export NEEDLE="DbName"
+            ;;
+        "dbuser")
+            export NEEDLE="DbUser"
+            ;;
+        "dbpassword")
+            export NEEDLE="DbPassword"
+            ;;
+        *)
+            if [ -n "${DEFAULT}" ]; then
+                echo "${DEFAULT}"
+                return 0
+            fi
+            ;;
+    esac
+    # Retrieve NEEDLE line in catalog section output
+    # For testing excluding -u, -g and adding -c    
+    if ! VALUE="$(${BAREOS_DIRECTOR_BINARY} -c "${BAREOS_CONFIG_DIR}" -xc | \
+            awk '/Catalog \{/ {p=1; next}; /\}/ {p=0}; {if (p==1){ if ($0 ~ " "ENVIRON["NEEDLE"]" =" ) print $0}}' | \
+            sed -e "s/*[[:space:]]$//" -e "s/^\(.*\)\([[:space:]]=[[:space:]]\)\(.*\)$/\3/" -e "s/\"//g" \
```

Comment:
This shell construct is quite complication. Even I have problems to understand what is does. I implemented another approach. Please consider if you want to integrate my last commits from https://github.com/joergsteffens/bareos/commits/dev/joergs/master/configdump